lib/msal-browser/src/broker/nativeBroker/PlatformAuthDOMHandler.ts

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    Logger,
+    createAuthError,
+    AuthErrorCodes,
+    IPerformanceClient,
+    StringDict,
+} from "@azure/msal-common/browser";
+import {
+    PlatformBrokerRequest,
+    PlatformDOMTokenRequest,
+} from "./PlatformBrokerRequest.js";
+import { createNewGuid } from "../../crypto/BrowserCrypto.js";
+import { NativeConstants } from "../../utils/BrowserConstants.js";
+import {
+    PlatformBrokerResponse,
+    PlatformDOMTokenResponse,
+} from "./PlatformBrokerResponse.js";
+import { createNativeAuthError } from "../../error/NativeAuthError.js";
+import { IPlatformAuthHandler } from "./IPlatformAuthHandler.js";
+
+export class PlatformAuthDOMHandler implements IPlatformAuthHandler {
+    protected logger: Logger;
+    protected performanceClient: IPerformanceClient;
+    protected correlationId: string;
+    protected brokerId: string;
+    protected extensionVersion: string;
+    platformAuthType: string;
+
+    constructor(
+        logger: Logger,
+        performanceClient: IPerformanceClient,
+        correlationId?: string
+    ) {
+        this.logger = logger;
+        this.performanceClient = performanceClient;
+        this.brokerId = NativeConstants.MICROSOFT_ENTRA_BROKERID;
+        this.correlationId = correlationId || createNewGuid();
+        this.extensionVersion = "";
+        this.platformAuthType = NativeConstants.PLATFORM_DOM_PROVIDER;
+    }
+
+    static async createProvider(
+        logger: Logger,
+        performanceClient: IPerformanceClient
+    ): Promise<PlatformAuthDOMHandler | undefined> {
+        logger.trace("PlatformAuthDOMHandler: createProvider called");
+
+        // @ts-ignore
+        if (window.navigator?.platformAuthentication) {
+            const supportedContracts =
+                // @ts-ignore
+                await window.navigator.platformAuthentication.getSupportedContracts(
+                    NativeConstants.MICROSOFT_ENTRA_BROKERID
+                );
+            if (
+                supportedContracts?.includes(NativeConstants.PLATFORM_DOM_APIS)
+            ) {
+                logger.trace("Platform auth api available in DOM");
+                return new PlatformAuthDOMHandler(
+                    logger,
+                    performanceClient,
+                    NativeConstants.MICROSOFT_ENTRA_BROKERID
+                );
+            }
+        }
+        return undefined;
+    }
+
+    /**
+     * Returns the Id for the broker extension this handler is communicating with
+     * @returns
+     */
+    getExtensionId(): string {
+        return this.brokerId;
+    }
+
+    /**
+     * Gets the version of the browser this handler is communicating with
+     */
+    getExtensionVersion(): string | undefined {
+        return "";
+    }
+
+    getExtensionName(): string | undefined {
+        return this.brokerId;
```

Comment:
Should this be something like "DOM API" instead since we already have id above ^?